Build failure on GCC 10

OS: Fedora 32
Compiler: GCC 10.2.1

Build log:
```c++
FAILED: CMakeFiles/libabsl.dir/src/third_party/abseil-cpp/absl/strings/cord.cc.o 
/usr/bin/c++  -DHAVE_NETINET_IN_H -I../src/third_party/abseil-cpp -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -DNDEBUG   -Wno-deprecated-declarations -Wno-attributes -Wno-narrowing -Wno-return-type -msse2 -std=gnu++2a -MD -MT CMakeFiles/libabsl.dir/src/third_party/abseil-cpp/absl/strings/cord.cc.o -MF CMakeFiles/libabsl.dir/src/third_party/abseil-cpp/absl/strings/cord.cc.o.d -o CMakeFiles/libabsl.dir/src/third_party/abseil-cpp/absl/strings/cord.cc.o -c ../src/third_party/abseil-cpp/absl/strings/cord.cc
In file included from ../src/third_party/abseil-cpp/absl/strings/cord.cc:33:
../src/third_party/abseil-cpp/absl/container/fixed_array.h: In instantiation of 'class absl::FixedArray<absl::cord_internal::CordRep*>':
../src/third_party/abseil-cpp/absl/strings/cord.cc:470:34:   required from here
../src/third_party/abseil-cpp/absl/container/fixed_array.h:110:9: error: no type named 'pointer' in 'using allocator_type = using allocator_type = class std::allocator<absl::cord_internal::CordRep*>' {aka 'class std::allocator<absl::cord_internal::CordRep*>'}
  110 |   using pointer = typename allocator_type::pointer;
      |         ^~~~~~~
../src/third_party/abseil-cpp/absl/container/fixed_array.h:111:9: error: no type named 'const_pointer' in 'using allocator_type = using allocator_type = class std::allocator<absl::cord_internal::CordRep*>' {aka 'class std::allocator<absl::cord_internal::CordRep*>'}
  111 |   using const_pointer = typename allocator_type::const_pointer;
      |         ^~~~~~~~~~~~~
../src/third_party/abseil-cpp/absl/container/fixed_array.h:112:9: error: no type named 'reference' in 'using allocator_type = using allocator_type = class std::allocator<absl::cord_internal::CordRep*>' {aka 'class std::allocator<absl::cord_internal::CordRep*>'}
  112 |   using reference = typename allocator_type::reference;
      |         ^~~~~~~~~
../src/third_party/abseil-cpp/absl/container/fixed_array.h:113:9: error: no type named 'const_reference' in 'using allocator_type = using allocator_type = class std::allocator<absl::cord_internal::CordRep*>' {aka 'class std::allocator<absl::cord_internal::CordRep*>'}
  113 |   using const_reference = typename allocator_type::const_reference;
      |         ^~~~~~~~~~~~~~~
../src/third_party/abseil-cpp/absl/container/fixed_array.h:118:9: error: no type named 'pointer' in 'using allocator_type = class std::allocator<absl::cord_internal::CordRep*>' {aka 'class std::allocator<absl::cord_internal::CordRep*>'}
  118 |   using reverse_iterator = std::reverse_iterator<iterator>;
      |         ^~~~~~~~~~~~~~~~
../src/third_party/abseil-cpp/absl/container/fixed_array.h:119:9: error: no type named 'const_pointer' in 'using allocator_type = class std::allocator<absl::cord_internal::CordRep*>' {aka 'class std::allocator<absl::cord_internal::CordRep*>'}
  119 |   using const_reverse_iterator = std::reverse_iterator<const_iterator>;
      |         ^~~~~~~~~~~~~~~~~~~~~~
../src/third_party/abseil-cpp/absl/strings/cord.cc: In function 'absl::cord_internal::CordRep* absl::NewTree(const char*, size_t, size_t)':
../src/third_party/abseil-cpp/absl/strings/cord.cc:477:9: error: no match for 'operator[]' (operand types are 'absl::FixedArray<absl::cord_internal::CordRep*>' and 'size_t' {aka 'long unsigned int'})
  477 |     reps[n++] = VerifyTree(rep);
      |         ^
../src/third_party/abseil-cpp/absl/strings/cord.cc:481:32: error: 'class absl::FixedArray<absl::cord_internal::CordRep*>' has no member named 'data'
  481 |   return MakeBalancedTree(reps.data(), n);
      |                                ^~~~
../src/third_party/abseil-cpp/absl/strings/cord.cc: At global scope:
../src/third_party/abseil-cpp/absl/strings/cord.cc:429:17: warning: 'absl::cord_internal::CordRep* absl::MakeBalancedTree(absl::cord_internal::CordRep**, size_t)' defined but not used [-Wunused-function]
  429 | static CordRep* MakeBalancedTree(CordRep** reps, size_t n) {
      |                 ^~~~~~~~~~~~~~~~

```

--- a/cmake/init_target.cmake
+++ b/cmake/init_target.cmake
@@ -5,12 +5,12 @@
 # https://github.com/desktop-app/legal/blob/master/LEGAL
 
 function(init_target target_name) # init_target(my_target folder_name)
-    if (WIN32)
-        target_compile_features(${target_name} PUBLIC cxx_std_17)
-    elseif (APPLE)
+    if (APPLE)
         target_compile_features(${target_name} PUBLIC cxx_std_14)
     else()
-        target_compile_features(${target_name} PUBLIC cxx_std_20)
+        # C++20 is not supported by bundled abseil-cpp yet:
+        # https://github.com/abseil/abseil-cpp/issues/722
+        target_compile_features(${target_name} PUBLIC cxx_std_17)
     endif()
     if (CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
         set_target_properties(${target_name} PROPERTIES

